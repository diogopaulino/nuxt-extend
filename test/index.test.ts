import config from './fixture/app/nuxt.config'

const scrub = (input) => {
  if (typeof input === 'string') {
    return input.replace(__dirname, '{test}')
  }

  if (Array.isArray(input)) {
    return input.map(i => scrub(i))
  }

  if (typeof input === 'object') {
    const res = {}
    for (const key in input) {
      res[key] = scrub(input[key])
    }
    return res
  }

  return input
}

it('matches snapshot', () => {
  expect(scrub(config)).toMatchSnapshot()
})
