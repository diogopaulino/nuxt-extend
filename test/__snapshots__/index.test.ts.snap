// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "_level": 0,
  "alias": Object {
    "@@baseTheme": "{test}/fixture/themes/base",
    "@@baseThemeEx": "{test}/fixture/themes/red",
    "@baseTheme": "{test}/fixture/themes/base",
    "@baseThemeEx": "{test}/fixture/themes/base",
    "~baseTheme": "{test}/fixture/themes/base",
    "~baseThemeEx": "{test}/fixture/themes/base",
    "~~baseTheme": "{test}/fixture/themes/base",
    "~~baseThemeEx": "{test}/fixture/themes/red",
  },
  "components": Array [
    Object {
      "level": 0,
      "path": "{test}/fixture/app/components",
    },
    Object {
      "level": 1,
      "path": "{test}/fixture/themes/red/components",
    },
    Object {
      "global": true,
      "level": 2,
      "path": "{test}/fixture/themes/base/components",
    },
  ],
  "extends": "../themes/red/nuxt.config",
  "hooks": Object {},
  "name": "myApp",
  "publicRuntimeConfig": Object {
    "color": "red",
  },
  "rootDir": "{test}/fixture/app",
  "srcDir": "{test}/fixture/themes/base",
}
`;
